test/e2e/e2e_setup_test.go

```diff
@@ -224,15 +286,16 @@ func Setup(specName string, f func(testSpecificSettings func() testSettings), op
 				}
 				return flavor
 			},
-			RuntimeExtensionProviders: runtimeExtensionProviders,
+			RuntimeExtensionProviders:   runtimeExtensionProviders,
+			UseKindForManagementCluster: options.additionalVCSimServer,
```

Comment:
Why is UseKindForManagementCluster set to additionalVCSimServer? Shouldn't this be an entirely separate field?

Reply:
Also does this mean we now have cases again where we can't use kind for the additional management cluster?

Reply:
We cannot create two VCSimServers in-parallel at the same management cluster. So `additionalVCSimServer` only makes sense to be used when we have a separate management cluster.

We don't block other cases because we are not forced to use the testsettings getter.

I added it here to follow https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3109#discussion_r1683333712

I'm also fine to not expose the `UseKindForManagementCluster` bool and just explicitly set it.